src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -1574,12 +1606,12 @@ private void HandleDirective(DirectiveDescriptor descriptor)
                         break;
 
                     case DirectiveTokenKind.Namespace:
-                        if (!QualifiedIdentifier())
+                        if (!QualifiedIdentifier(out var identifierLength))
                         {
                             Context.ErrorSink.OnError(
                                 CurrentStart,
                                 LegacyResources.FormatDirectiveExpectsNamespace(descriptor.Name),
-                                CurrentSymbol.Content.Length);
+                                identifierLength);
```

Comment:
This is a really strange way to do things, but OK. We'll revisit when we do error recovery.